django/db/migrations/serializer.py

```diff
@@ -244,8 +244,9 @@ def _format(self):
 
 class SetSerializer(BaseSequenceSerializer):
     def _format(self):
-        # Don't use the literal "{%s}" as it doesn't support empty set
-        return "set([%s])"
+        # When value is empty, the empty set should be serialized as "set()",
+        # not "{}" because {} is an empty dict.
+        return '{%s}' if self.value else 'set(%s)'
```

Comment:
We should remember that migration files are auto generated code. They first and foremost need to not have syntax errors.

There was a discussion on running some form of isort on migration files while generating them. We decided against that for the sake of migration files being auto-generated. Furthermore, lines in migration files can easily end up having lines way past 79 or 119 characters. And I'm against starting to linebreak those lines with some other tool. Users should exclude migration files from linting, IMO.